### PR TITLE
PIM-4708: Handle locale specific feature in the new PEF

### PR DESCRIPTION
**Pull request by grena**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | Y
| Specs & Static     | Runnin'
| Changelog updated  | N

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php [hunk 1]
```diff
@@ -117,12 +117,19 @@ protected function filterValuesData(array $valuesData)
                 $newContextValues = [];
 
                 foreach ($contextValues as $contextValue) {
-                    if (null === $contextValue['locale'] ||
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: rename it to $values

File: src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php [hunk 2]
```diff
@@ -117,12 +117,19 @@ protected function filterValuesData(array $valuesData)
...
-                        !$this->objectFilter->filterObject(
-                            $this->getLocale($contextValue['locale']),
-                            'pim.internal_api.locale.edit'
-                        )
-                    ) {
+                    $hasLocale = null !== $contextValue['locale'];
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Filter/ProductEditDataFilter.php]: rename to $isLocalizable

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 3]
```diff
@@ -187,6 +193,11 @@ define([
 
                 productValue.data = value;
                 mediator.trigger('pim_enrich:form:entity:update_state');
+            },
+            getLabel: function () {
+                return !!this.attribute.label[this.context.uiLocale] ?
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: useless cast

**Comment by phaseinducer** [comment]
jsdoc ! =)

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 4]
```diff
@@ -78,10 +99,28 @@ define([
...
+             * @throws {Error} if this method is not implemented
+             */
             renderInput: function () {
                 throw new Error('You should implement your field template');
             },
+
+            /**
+             * Is called after rendering the input
+             */
             postRender: function () {},
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: Is it still used ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 5]
```diff
@@ -123,41 +178,108 @@ define([
 
                 this.model.set('values', values);
             },
+
+            /**
+             * The the context of this field
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: *set ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 6]
```diff
@@ -123,41 +178,108 @@ define([
...
+             *
+             * @param {Object} context
+             */
             setContext: function (context) {
                 this.context = context;
             },
+
+            /**
+             * Add element to this field block
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: add **an** element

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 7]
```diff
@@ -123,41 +178,108 @@ define([
...
+             * Return whether is valid
+             *
+             * @returns {boolean}
+             */
             isValid: function () {
                 return this.valid;
             },
+
+            /**
+             * Set focus the input of this field
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: **the** focus **on**

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 8]
```diff
@@ -123,41 +178,108 @@ define([
...
+             */
             setFocus: function () {
                 this.$('input:first').focus();
             },
+
+            /**
+             * Set editable this field
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: Set this field as editable ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 9]
```diff
@@ -123,41 +178,108 @@ define([
...
+             *
+             * @param editable
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: {boolean}

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 10]
```diff
@@ -123,41 +178,108 @@ define([
...
+             * Return whether this field is editable
+             *
+             * @returns {boolean}
+             */
             isEditable: function () {
                 return this.editable;
             },
+
+            /**
+             * Set ready this field
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: Set this field as ready ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js [hunk 11]
```diff
@@ -165,15 +287,34 @@ define([
...
+            inputCanBeSeen: function (context) {
+                return this.attribute.is_locale_specific ?
+                    _.contains(this.attribute.locale_specific_codes, context.locale) :
+                    true;
+            },
+
+            /**
+             * Get current model value for this field
+             *
+             * @returns {*}
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: the returned value will always be an {Object}

Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: The returned value is an Object : `{locale: 'en_US', scope: null, data: 'stuff'}`

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js (hunk 4 above)
Comment (grena) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/field.js]: Yep in many field types (date, multiselect, etc)

**Comment by phaseinducer** [comment]
GTM if CI is ok

**Comment by sumbobyboys** [comment]
GTM !

**Pull request merged**